tools/make_class_power.py

```diff
@@ -184,55 +184,32 @@ def make_class_power(paramfile, external_pk = None, extraz=None):
         for red in extraz:
             trans = powspec.get_transfer(z=red)
             tfile = os.path.join(sdir, config['FileWithTransferFunction']+"-"+str(red))
-            save_transfer(trans, tfile, bg, red)
+            save_transfer(trans, tfile)
+            trans['k'][-1] *= 0.9999
             #Get and save the matter power spectrum
             pk_lin = powspec.get_pklin(k=trans['k'], z=red)
             pkfile = os.path.join(sdir, config['FileWithInputSpectrum']+"-"+str(red))
             if os.path.exists(pkfile):
                 raise IOError("Refusing to write to existing file: ",pkfile)
             np.savetxt(pkfile, np.vstack([trans['k'], pk_lin]).T)
 
-def save_transfer(transfer, transferfile, bg, redshift):
-    """Save a transfer function. Note we save the CAMB FORMATTED transfer functions.
-    These can be generated from CLASS by passing the 'format = camb' on the command line.
-    The transfer functions differ by:
+def save_transfer(transfer, transferfile):
+    """Save a transfer function. Note we save the CLASS FORMATTED transfer functions.
+    The transfer functions differ from CAMB by:
         T_CAMB(k) = -T_CLASS(k)/k^2 """
     if os.path.exists(transferfile):
```

Comment:
The check shall probably be done in the main function rather than here -- the current function really is maybe_save_transfer.

Reply:
Fixed!